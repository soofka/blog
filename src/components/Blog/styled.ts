import styledComponents from 'styled-components';

export const StyledBlog = styledComponents.div`
  width: 100%;
  height: 100%;
  font-size: ${props => props.theme.baseFontSize};
  line-height: ${props => props.theme.baseLineHeight};
`;

export const StyledBlogContainer = styledComponents.div`
  position: relative;
  width: 100%;
  max-width: ${props => props.theme.siteMaxWidth};
  height: 100%;
  margin: 0 auto;
`;
