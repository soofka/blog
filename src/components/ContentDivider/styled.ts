import styledComponents from 'styled-components';

export const StyledContentDivider = styledComponents.hr`
  border-color: ${props => props.theme.lighterBlue} !important;
`;

export default StyledContentDivider;
