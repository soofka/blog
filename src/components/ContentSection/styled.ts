import styledComponents from 'styled-components';

export const StyledContentSection = styledComponents.div`
  float: left;
  margin: 0 0 0 ${props => props.theme.infoSectionWidth};
  padding: ${props => props.theme.paddingM};
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: ${props => props.theme.lighterBlack};
  background-color: ${props => props.theme.white};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.white}`};

  a {
    color: ${props => props.theme.lighterBlack};

    &:hover,
    &:active {
      color: ${props => props.theme.black};
    }
  }

@media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
  margin: 0;
  padding: ${props => props.theme.paddingS};
  width: 100%;
}`;

export default StyledContentSection;
