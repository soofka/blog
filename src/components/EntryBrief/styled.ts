import styledComponents from 'styled-components';

export const StyledEntryBrief = styledComponents.div`
  float: left;
  width: 100%;
  height: ${props => 24 * props.theme.baseDistance}px;
  margin: ${props => props.theme.marginS} 0 ${props => props.theme.marginM} 0;
  padding: 0;
  border: 2px solid ${props => props.theme.white};
  outline: 1px solid ${props => props.theme.lighterBlack};
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: scroll;
  background-size: cover;
`;

export default StyledEntryBrief;
