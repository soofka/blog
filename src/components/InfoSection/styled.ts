import styledComponents from 'styled-components';

export const StyledInfoSection = styledComponents.div`
  position: static;
  display: inline-block;
  width: 100%;
  height: ${props => props.theme.infoSectionHeightValue}px;
  color: ${props => props.theme.darkerWhite};
  background-color: ${props => props.theme.blue};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${props => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${props => props.theme.white};
    }
  }
`;

// export const StyledInfoSection = styledComponents.div`
//   position: absolute;
//   top: 0;
//   left: -${props => props.theme.infoSectionWidthValue}px;
//   box-sizing: border-box;
//   width: ${props => props.theme.infoSectionWidthValue}px;
//   height: auto;
//   min-height: 100%;
//   padding: ${props => props.theme.paddingS};
//   color: ${props => props.theme.darkerWhite};
//   background-color: ${props => props.theme.blue};
//   box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};
//
//   a {
//     color: ${props => props.theme.darkerWhite};
//     text-decoration: none;
//
//     &:hover,
//     &:active {
//       color: ${props => props.theme.white};
//     }
//   }
//
//   @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
//     position: static;
//     display: inline-block;
//     width: 100%;
//     height: ${props => props.theme.infoSectionMobileHeightValue}px;
//     min-height: 0;
// }
// `;

export default StyledInfoSection;

