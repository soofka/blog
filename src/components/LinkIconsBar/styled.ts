import styledComponents from 'styled-components';

export const StyledLinkIconsBar = styledComponents.ul`
  position: absolute;
  left: 0;
  bottom 0;
  width: 100%;
  margin: ${props => props.theme.marginM} 0;
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    padding: 0;

    svg {
      color: ${props => props.theme.white};
      width: ${props => 6 * props.theme.baseDistanceValue}px;
      height: ${props => 6 * props.theme.baseDistanceValue}px;
      
      opacity: 0.7;

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }
`;

export default StyledLinkIconsBar;
