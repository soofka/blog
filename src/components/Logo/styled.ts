import styledComponents from 'styled-components';

export const StyledLogoImageContainer = styledComponents.span`
  float: left;
  margin: ${props => props.theme.marginS};
`;

export const StyledLogoImage = styledComponents.img`
  width: ${props => 12 * props.theme.baseDistanceValue}px;
  height: auto;
  border-radius: ${props => 4 * props.theme.baseDistanceValue}px;
  
  @media (max-width: ${props => props.theme.breakpointM}) {
    width: ${props => 8 * props.theme.baseDistanceValue}px;
    border-radius: ${props => 2 * props.theme.baseDistanceValue}px;
  }
`;

export const StyledLogoHeader = styledComponents.h1`
  float: left;
  font-family: ${props => props.theme.headerFontFamily};
  margin: ${props => props.theme.marginXL};
  
  @media (max-width: ${props => props.theme.breakpointM}) {
    margin: ${props => props.theme.marginL};
  }
`;
