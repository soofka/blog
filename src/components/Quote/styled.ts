import styledComponents from 'styled-components';

export const StyledQuote = styledComponents.p`
  margin: ${props => props.theme.marginM};
  padding: ${props => props.theme.paddingS};
  font-style: italic;
  text-align: center;

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    display: none;
  }
`;

export default StyledQuote;
