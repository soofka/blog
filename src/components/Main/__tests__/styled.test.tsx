import * as React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import theme from 'common/theme';
import { StyledMain } from 'components/Main/styled';

describe('<StyledMain/>', () => {

  it('is styled correctly', () => {
    expect(shallow(<StyledMain theme={theme} />)).toMatchSnapshot();
  });

});
