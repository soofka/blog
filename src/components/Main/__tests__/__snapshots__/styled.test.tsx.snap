// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledContentSection/> is styled correctly 1`] = `
.c0 {
  float: left;
  margin: 0 0 0;
  padding: 16px;
  width: 100%;
  height: auto;
  min-height: 100%;
  box-sizing: border-box;
  color: rgb(44,44,44);
  background-color: rgb(246,246,246);
  box-shadow: 0 0 8px 0 rgb(246,246,246);
}

.c0 a {
  color: rgb(44,44,44);
}

.c0 a:hover,
.c0 a:active {
  color: rgb(20,20,20);
}

@media (max-width:968px) {
  .c0 {
    margin: 0;
    padding: 8px;
    width: 100%;
  }
}

<div
  className="c0"
/>
`;
